src/corecel/Macros.hh

```diff
@@ -42,6 +42,33 @@
 #    define CELER_FORCEINLINE inline
 #endif
 
+#define CELER_COMP_UNKNOWN 0
+#define CELER_COMP_CLANG 1
+#define CELER_COMP_GCC 2
+#define CELER_COMP_INTEL 3
+#define CELER_COMP_MSVC 4
+#define CELER_COMP_NVCPP 5
+
+/*!
+ * \def CELER_COMP_HOST
+ *
+ * Set to the value of one of the CELER_COMP_<compiler> macros.
+ * Allows checking if a specific compiler is used.
+ */
+#if defined(__clang__)
+#    define CELER_COMP_HOST CELER_COMP_CLANG
```

Comment:
I don't think `CELER_COMP_HOST` is right, since the compilation phase could be for "device" as well. Maybe it should just be `CELER_COMPILER` (and `CELER_COMPILER_CLANG` etc.)? Also there are starting to be more "clang derivative" compilers (e.g intel, rocm), and is this defined then as well?

Reply:
By clang derivative do you mean llvm-based compilers? You have `__llvm__` for that. I didn't check if they define `__clang__` as well.

Reply:
Ah ok good point. Maybe though, until we have use cases for the other compilers and check that the macros work as intended for those, we should just leave the other options out?

Reply:
sure, I'll just add clang for now